cmake/third_party/azure_sdk.cmake

```diff
@@ -0,0 +1,81 @@
+include(GNUInstallDirs)
+include(${CMAKE_SOURCE_DIR}/cmake/helpers.cmake)
+cmake_minimum_required(VERSION 3.13)
+
+# Skip the AZURE SDK build step if the extension is not enabled.
+if(NOT ENABLE_AZURE)
+    return()
+endif()
+
+config_choice(
+    IMPORT_AZURE_SDK
+    "Specify how to import the AZURE SDK"
+    OPTIONS
+        "none;IMPORT_AZURE_SDK_NONE;NOT ENABLE_AZURE"
+        "package;IMPORT_AZURE_SDK_PACKAGE;ENABLE_AZURE"
+        "external;IMPORT_AZURE_SDK_EXTERNAL;ENABLE_AZURE"
+)
+ 
+if(IMPORT_AZURE_SDK_NONE)
+    message(FATAL_ERROR "Cannot enable Azure extension without specifying an IMPORT_AZURE_SDK method (package, external).")
+endif()
+
+set(azure_storage_lib_location)
+set(azure_core_lib_location)
+set(azure_storage_common_lib_location)
+set(azure_sdk_include_location)
+
+if(IMPORT_AZURE_SDK_PACKAGE)
+    message(FATAL_ERROR "Currently the package config option is not implement yet.")
+endif()
+
+if (IMPORT_AZURE_SDK_EXTERNAL)
+    ExternalProject_Add(
+        azure-sdk
+        PREFIX azure-sdk-cpp
+        GIT_REPOSITORY      https://github.com/Azure/azure-sdk-for-cpp.git
+        GIT_TAG             azure-storage-blobs_12.2.0
+        CMAKE_ARGS
+            -DBUILD_SHARED_LIBS=ON
+            -DCMAKE_POSITION_INDEPENDENT_CODE=ON
+            -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/azure-sdk-cpp/install
+        BUILD_ALWAYS FALSE
+        BUILD_BYPRODUCTS
+            ${CMAKE_CURRENT_BINARY_DIR}/azure-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libazure-storage-blobs${CMAKE_SHARED_LIBRARY_SUFFIX}
+            ${CMAKE_CURRENT_BINARY_DIR}/azure-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libazure-core${CMAKE_SHARED_LIBRARY_SUFFIX}
+	    ${CMAKE_CURRENT_BINARY_DIR}/azure-sdk-cpp/install/${CMAKE_INSTALL_LIBDIR}/libazure-storage-common${CMAKE_SHARED_LIBRARY_SUFFIX}
+        INSTALL_DIR ${CMAKE_CURRENT_BINARY_DIR}/azure-sdk-cpp/install
+        TEST_COMMAND ""
+        UPDATE_COMMAND ""
+    )
+    ExternalProject_Get_Property(azure-sdk INSTALL_DIR)
+    file(MAKE_DIRECTORY ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
+
+    # Set the path variables to be used for the AZURE targets.
+    set(azure_sdk_include_location ${INSTALL_DIR}/${CMAKE_INSTALL_INCLUDEDIR})
+    set(azure_storage_lib_location ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libazure-storage-blobs${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(azure_core_lib_location ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libazure-core${CMAKE_SHARED_LIBRARY_SUFFIX})
+    set(azure_storage_common_lib_location ${INSTALL_DIR}/${CMAKE_INSTALL_LIBDIR}/libazure-storage-common${CMAKE_SHARED_LIBRARY_SUFFIX})
+endif()
+
+add_library(azure_storage_lib SHARED IMPORTED)
+add_library(azure_core_lib SHARED IMPORTED)
+add_library(azure_storage_common_lib SHARED IMPORTED)
+
+# Small workaround to declare the include directory under INTERFACE_INCLUDE_DIRECTORIES during the configuration phase.
```

Comment:
Could you elaborate on what this workaround does, and what fails if you don't include it? Just from reading the comment it's unclear to me why we need to do this.

Reply:
I'm changing the comment to "Declare the include directories under INTERFACE_INCLUDE_DIRECTORIES during the configuration phase." to make things clearer

Reply:
lmk if i need to add more detail https://github.com/wiredtiger/wiredtiger/pull/8702/commits/f7f492559b85b0eab5dda6e4def51d6870a4eb94

Reply:
I'm still not sure what this means, sorry - in general with comments, it's better to write down "why" than "how", since the code already says "how". What breaks if we don't do this?

Reply:
Sorry - yup this step is needed to set the imported locations after we have declared the include directories as shared imported earlier. If we do not set the location the build fails expecting a location.

`Declare the include directories under INTERFACE_INCLUDE_DIRECTORIES during the configuration phase to set the IMPORTED_LOCATION for imported targets`

Reply:
Sorry to be a pain, but - this still doesn't explain *why* we do this. (i.e. why must these three libraries have these properties, and no others?) There's every chance I'm missing something though, so feel free to push back.

Reply:
add_library() are basically variable declarations and are shared libraries that can be used by multiple libraries all at once. These are the 3 libraries that we are building from the azure sdk that we have cloned earlier.

Reply:
updated comment https://github.com/wiredtiger/wiredtiger/pull/8702/commits/6485442efe332b2f7cacc27b9640cbb623869cc5